frontend/src/scenes/events/eventsTableLogic.test.ts

```diff
@@ -0,0 +1,494 @@
+            orderBy: '-timestamp',
+            selectedEvent: null,
+            newEvents: [],
+            highlightEvents: {},
+            columnConfigSaving: false,
+            automaticLoadEnabled: false,
+            columnConfig: 'DEFAULT',
+        })
+    })
+
+    describe('reducers', () => {
+        it('can toggle autoloading on', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.toggleAutomaticLoad(true)
+            }).toMatchValues({
+                automaticLoadEnabled: true,
+            })
+        })
+
+        it('can toggle autoloading on and off', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.toggleAutomaticLoad(true)
+                logic.actions.toggleAutomaticLoad(false)
+            }).toMatchValues({
+                automaticLoadEnabled: false,
+            })
+        })
+
+        it('does not call prependNewEvents when there are zero new events', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.toggleAutomaticLoad(true)
+            }).toMatchValues({
+                automaticLoadEnabled: true,
+            })
+        })
+
+        it('does call prependNewEvents when there are new events', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.pollEventsSuccess([makeEvent(), makeEvent(), makeEvent()])
+                logic.actions.toggleAutomaticLoad(true)
+            })
+                .toMatchValues({
+                    automaticLoadEnabled: true,
+                })
+                .toDispatchActions(['prependNewEvents'])
+        })
+
+        it('does not call prependNewEvents when there are new events but loading is off', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.pollEventsSuccess([makeEvent(), makeEvent(), makeEvent()])
+                logic.actions.toggleAutomaticLoad(false)
+            }).toMatchValues({
+                automaticLoadEnabled: false,
+            })
+        })
+
+        it('can mark that column config is saving', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.setColumnConfigSaving(true)
+            }).toMatchValues({
+                columnConfigSaving: true,
+            })
+        })
+
+        it('can set a poll timeout ID', async () => {
+            const timeoutHandle = setTimeout(() => {})
+            await expectLogic(logic, () => {
+                logic.actions.setPollTimeout(timeoutHandle)
+            }).toMatchValues({
+                pollTimeout: timeoutHandle,
+            })
+        })
+
+        it('can highlight new events', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.prependNewEvents([makeEvent('potato')])
+            }).toMatchValues({
+                highlightEvents: { potato: true },
+            })
+        })
+
+        //TODO why does it?
```

Comment:
Lazy timing probably? I'd rather just clear these after a second or so.

Reply:
Ah, I'd missed that TODO. There was no need to manually clear them 🙌